src/app/store/onboarding/onboarding.slice.ts

```diff
@@ -28,13 +28,13 @@ export const onboardingSlice = createSlice({
   initialState,
   reducers: {
     hideSuggestedFirstSteps(state, action: PayloadAction<boolean>) {
-      state.hasHiddenSuggestedFirstSteps = action.payload;
+      state.hideSteps = action.payload;
     },
     skipFundAccount(state, action: PayloadAction<boolean>) {
       state.hasSkippedFundAccount = action.payload;
     },
-    updateSuggestedFirstStepsStatus(state, action: PayloadAction<SuggestedFirstStepsStatus>) {
-      state.suggestedFirstStepsStatus = action.payload;
+    updateSuggestedFirstStepsStatus(state, action: PayloadAction<Record<string, number>>) {
```

Comment:
```suggestion
    updateSuggestedFirstStepsStatus(state, action: PayloadAction<Record<SuggestedFirstSteps, SuggestedFirstStepStatus>>) {
```

Reply:
I tried that before, why do I get this?

![Screen Shot 2022-06-02 at 5 24 36 AM](https://user-images.githubusercontent.com/6493321/171610499-e121049a-0281-4d1f-8e7c-8d0218198557.png)

Reply:
It looks like this type means the whole object, with all keys, but we can use `Partial`  

```ts
PayloadAction<Partial<Record<SuggestedFirstSteps, SuggestedFirstStepStatus>>>
```

A complex-ish type, but more accurate than `string`. 

------------------------

I think the pattern used here kind of violates the setters principal, though https://redux.js.org/style-guide/#model-actions-as-events-not-setters

> Whatever this object is, just spread it onto this this object's state

If we were to think of this state change as an event, it's just the completion of a step. We never set a step to incomplete (beyond the default), so we can be more explicit. 

Rather than `updateSuggestedFirstStepStatus`, what if we had a `userCompletedSuggestedStep` action, that only takes a step, and sets it to done?  


```ts
    userCompletedSuggestedStep(state, action: PayloadAction<{ step: SuggestedFirstSteps }>) {
      state.stepsStatus[action.payload.step] = SuggestedFirstStepStatus.Done;
    },
```

This tidies up the rest of the code quite a lot

Reply:
I agree this is better, but wouldn't this require a migration?

Reply:
O nvm, it does not.